### PR TITLE
check if lastWriteTime is valid

### PR DESCRIPTION
**Pull request by campersau**
otherwise an exception is thrown

```
System.ArgumentOutOfRangeException
  HResult=0x80131502
  Message=Ungültige Win32-FileTime.
  Source=mscorlib
  StackTrace:
   at System.DateTime.ToFileTimeUtc()
```

### CODE REVIEW

File: PackageExplorer/Interop/NativeDragDrop.cs [hunk 1]
```diff
@@ -22,15 +22,18 @@ public static Stream CreateFileGroupDescriptorW(string fileName, DateTimeOffset
...
-            var changeTimeFileTime = new System.Runtime.InteropServices.ComTypes.FILETIME
+            if (lastWriteTime != default(DateTimeOffset))
             {
-                dwLowDateTime = (int)(changeTime & 0xffffffff),
-                dwHighDateTime = (int)(changeTime >> 32),
-            };
-            fileDescriptor.ftLastWriteTime = changeTimeFileTime;
-            fileDescriptor.ftCreationTime = changeTimeFileTime;
+                fileDescriptor.dwFlags |= FD_CREATETIME | FD_WRITESTIME;
+                var changeTime = lastWriteTime.ToLocalTime().ToFileTime();
```
Comment (onovotny) [inline PackageExplorer/Interop/NativeDragDrop.cs]: Should this be local time or UTC for this API?

Comment (campersau) [reply PackageExplorer/Interop/NativeDragDrop.cs]: That's some of the code I copied from http://dlaa.me/blog/post/9917797.

But looking at the source code of `DateTimeOffset.ToFileTime()` the call to `ToLocalTime()` is acutally unnecessary.

https://github.com/dotnet/corefx/blob/aeda6ab039f8d6e6041379ab156d51410b679c13/src/Common/src/CoreLib/System/DateTimeOffset.cs#L711-L714
or
https://referencesource.microsoft.com/#mscorlib/system/datetimeoffset.cs,621

I will remove it.

**Comment by campersau** [comment]
Hmm.. the main reason why the last write time is not set in the first place is that the `PhysicalPackageFile` reads the last write date only if the [stream is accessed](https://github.com/NuGet/NuGet.Client/blob/90524ca5f2313702df2f29b72649c9611aa4b086/src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PhysicalPackageFile.cs#L73-L103).

So the actual fix would be to create a wrapper or just implement our own `PhysicalPackageFile` which reads the last write date if needed an not just when the stream is accessed.

**Comment by onovotny** [comment]
Either sounds good, whichever approach is simplest.

File: Core/Packages/DiskPackageFile.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.IO;
+
+namespace NuGetPe
+{
+    public class DiskPackageFile : PackageFileBase
```
Comment (campersau) [inline Core/Packages/DiskPackageFile.cs]: I wanted to avoid naming collisions... Other suggestions?

**Pull request merged**